### ISSUE TITLE
CA1823 incorrectly flags const used as a fixed buffer length

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn-analyzers#2724]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

Example: v2.9.4 (Latest)

#### Diagnostic ID

Example: [CA1823](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1823-avoid-unused-private-fields?view=vs-2019)

#### Repro steps

```C#
using System.Runtime.InteropServices;

class Program
{
    static void Main() { }

    private const int SomeLength = 16;

    [StructLayout(LayoutKind.Sequential)]
    internal unsafe struct SomeStruct
    {
        internal fixed byte FixedBuffer[SomeLength];
    }
}
```

#### Expected behavior
No warnings.

#### Actual behavior
The CA1823 analyzer flags the SomeLength const as being unused:
```
Program.cs(7,23,7,33): warning CA1823: Unused field 'SomeLength'.
```
even though obviously it is being used.

**Comment by mavasani** [comment]
@jinujoseph @vatsalyaagrawal Can we port this issue to Roslyn repo? The issue does not seem specific to this analyzer. Seems like there is no `IFieldReferenceOperation` being generated for `SomeLength` usage in `internal fixed byte FixedBuffer[SomeLength];`. Find All References also does not seem to find this reference to `SomeLength`, so it is very likely that the underlying compiler bound tree does not have a bound node representing this field reference.
